chia/full_node/full_node.py

```diff
@@ -1253,7 +1253,8 @@ async def add_block_batch(
                 if error is not None:
                     self.log.error(f"Error: {error}, Invalid block from peer: {peer.get_peer_logging()} ")
                 return False, agg_state_change_summary, error
-            block_record = self.blockchain.block_record(block.header_hash)
+            block_record = await self.blockchain.get_block_record_from_db(block.header_hash)
```

Comment:
the record should be in the cache, im afraid this can hide/open some problems

Reply:
Do you mean that we should *add* code to populate the cache before getting here, or do you mean that there is already code to do this? (the tests would fail without being able to fall back to the DB here, so if there is logic that's meant to populate the cache, I don't think it's working right).

I haven't found code that looks like it attempts to make sure this block is in the cache.